src/EditorFeatures/CSharpTest/ValidateFormatString/ValidateFormatStringTests.cs

```diff
@@ -0,0 +1,945 @@
+        public async Task VerbatimMultipleLines()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(@""This {0} 
+{1} {2[||]} works"", ""multiple"", ""line"", ""test"")); 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task Interpolated()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        var Name = ""Peter"";
+        var Age = 30;
+       
+        string.Format($""{Name,[||] 20} is {Age:D3} ""); 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task Empty()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(""[||]""); 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task LeftParenOnly()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format([||]; 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task ParenthesesOnly()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format([||]); 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task EmptyString()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(""[||]""); 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task FormatOnly_NoStringDot()
+        {
+            await TestDiagnosticMissingAsync(@" using static System.String
+class Program
+{
+    static void Main(string[] args)
+    {
+        Format(""[||]""); 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task NamedParameters_BlankName()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format( : ""value""[||])); 
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task DuplicateNamedArgs()
+        {
+            await TestDiagnosticMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(format:""This [||] "", format:"" test ""); 
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task GenericIdentifier()
```

Comment:
I'm confused as to what this test is providing? Is it just that other methods named Format shouldn't trigger the analyzer? If so, then the generic doesn't contribute much, this should be a method with the same signature as a string.Format method.